keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepositoryTests.kt

```diff
@@ -301,6 +303,27 @@ abstract class ArtifactRepositoryTests<T : ArtifactRepository> : JUnit5Minutests
             get(ArtifactVersionStatus::previous).isEmpty()
           }
         }
+
+        test("an artifact version can be vetoed") {
```

Comment:
I debated putting "and the version has not been promoted to the environment" in the description, but it's already inside of the "No version has been promoted to an environment" context, so I felt that was redundant. But maybe it's better to be explicit?

Reply:
suggestion: maybe `an artifact version can be vetoed even if it was not previously deployed`?

Reply:
Fixed in 8400beb